USING_ELEKTRA.md

```diff
@@ -0,0 +1,136 @@
+
+Let's take the following example:
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' green
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+> green
+```
+
+Elektra's database is hierarchically structured which means that keys are organized 
+similar to directories in a file system. In the case
+of lcdproc, the relevant keys are located under `/sw/lcdproc/<app>/#0/current`
+where `<app>` will either be server configuration `lcdd` or the client configurations
+ `lcdproc`, `lcdvc` and `lcdexec`. LCDproc drivers are installed on the server, 
+ so in the upper example we used `lcdd` to change the background 
+ color of the curses driver to green.
+ 
+ The new lcdproc comes with a configuration specification such as [LCDd-spec.ini](server/specification/LCDd-spec.ini)
+ which is used by Elektra to validate configuration settings.
+ This guarantees that certain mistakes cannot occur like setting the background color to *greeen* which could
+ potentially crash the application. If you have followed the specification installation process correctly
+ from the [INSTALL.md](INSTALL.md), you now have the full specification [mounted](https://www.libelektra.org/tutorials/mount-configuration-files)
+ under the `spec` namespace. 
+ 
+ The specification for `.../curses/background` background for example
+ tells us that only `red, black, green, yellow, blue, magenta, cyan, white` are valid and that the value given 
+ has to be of type string. Elektra will prevent you to set invalid values which will help reduce errors and misconfiguration.
+  All these data and much more (eg. default values, descriptions, etc.) are saved as 
+ [metadata](https://www.libelektra.org/devdocu/metadata)
+ for each and every key under the `spec` namespace.
+ 
+ The following sections will teach you on how to inspect all relevant settings
+  and specifications and change desired configuration settings accordingly.
+ There best way to change settings is done via `kdb set`
+ since it integrates the best with specifications for configurations.
+ 
+### Changing Configurations
+ 
+Using the native command line is the best possibility to find out which configuration settings are available.
+If you want to see all available configuration settings for the `server`, you can simply call
+```sh
+kdb ls '/sw/lcdproc/lcdd/#0/current/server'
+#> /sw/lcdproc/lcdd/#0/current/server/autorotate
+#> /sw/lcdproc/lcdd/#0/current/server/backlight
+#> /sw/lcdproc/lcdd/#0/current/server/bind
+#> /sw/lcdproc/lcdd/#0/current/server/driver
+...
+```
+To see all available configuration settings at all you can execute `kdb ls '/sw/lcdproc/lcdd/#0/current'`.
+To see the current value you can simply call `kdb get` on the respective key.
+Now lets take the curses driver for example again and lets see what is the 
+current background color is:
+```sh
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> cyan
+```
+The result comes from the set `default` metadata which is saved for the key. 
```

Comment:
How do we know it comes from default? (`kdb get -v`)

Reply:
`kdb get -v` actually does not provide a good indication that the default value is taken. It confuses users even more.
```
kdb get -v '/sw/lcdproc/lcdd/#0/current/curses/foreground'
got 660 keys
searching spec/sw/lcdproc/lcdd/#0/current/curses/foreground, found: spec/sw/lcdproc/lcdd/#0/current/curses/foreground, options: KDB_O_CALLBACK
The resulting keyname is /sw/lcdproc/lcdd/#0/current/curses/foreground
The resulting value size is 5
blue
```

Reply:
Then please create a usability bug that the output is not helpful. (/ are cascading and at the same time default keys, maybe we should also change this "optimization").

Reply:
#2474